packages/rs-sdk/src/sdk.rs

```diff
@@ -185,6 +215,92 @@ impl Sdk {
         }
     }
 
+    /// Updates or fetches the nonce for a given identity and contract pair from a cache,
+    /// querying Platform if the cached value is stale or absent. Optionally
+    /// increments the nonce before storing it, based on the provided settings.
+    pub async fn get_identity_contract_nonce(
```

Comment:
This makes Sdk stateful.

The idea is that Sdk should be stateless. Stateful code should be implemented in `ContextProvider`.

Reply:
I agree that in general I don't like that SDK is stateful, but at the same time it's just a cache, the context provider is external data coming in right?

Reply:
Other things, like data contracts or quorum keys, are also "just a cache", and we do it in ContextProvider.

That's the design pattern we took when implementing rs-sdk - Sdk object is stateless, and stateful code belongs to ContextProvider.

See [GrpcContextProvider](https://github.com/dashpay/platform/blob/2c647f0fad75846abb05829062e17b3e69660fb2/packages/rs-sdk/src/mock/provider.rs#L153) for example.